sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -172,3 +172,93 @@ def test_binning_train_validation_are_separated():
+    (HistGradientBoostingRegressor, X_regression, y_regression)
+])
+def test_max_iter_with_warm_start_validation(GradientBoosting, X, y):
+    """Check that max_iter >= n_iter_ with warm starting.
+
+    Check that a ValueError is raised when the maximum number of iterations is
+    smaller than the number of iterations from the previous fit when warm_start
+    is True.
+
+    """
+    estimator = GradientBoosting(max_iter=50, warm_start=True)
+    estimator.fit(X, y)
+    estimator.set_params(max_iter=25)
+    err_msg = ('max_iter=25 must be larger than or equal to n_iter_=50 '
+               'when warm_start==True')
+    with pytest.raises(ValueError, match=err_msg):
+        estimator.fit(X, y)
+
+
+@pytest.mark.parametrize('GradientBoosting, X, y', [
+    (HistGradientBoostingClassifier, X_classification, y_classification),
+    (HistGradientBoostingRegressor, X_regression, y_regression)
+])
+def test_warm_start_yields_identical_results(GradientBoosting, X, y):
+    """Test that two fits with warm start yield the same results as one fit.
+
+    Make sure that fitting 50 iterations and then 25 with warm start is
+    equivalent to fitting 75 iterations.
+
+    """
+    rng = 42
+    gb_warm_start = GradientBoosting(
+        n_iter_no_change=100, max_iter=50, random_state=rng, warm_start=True
+    )
+    gb_warm_start.fit(X, y).set_params(max_iter=75).fit(X, y)
+
+    gb_no_warm_start = GradientBoosting(
+        n_iter_no_change=100, max_iter=75, random_state=rng, warm_start=False
+    )
+    gb_no_warm_start.fit(X, y)
+
+    # Check identical nodes for each tree
+    for (pred_ith_warm, pred_ith_no_warm) in zip(gb_warm_start._predictors,
+                                                 gb_no_warm_start._predictors):
+        for (predictor_warm, predictor_no_warm) in zip(pred_ith_warm,
+                                                       pred_ith_no_warm):
+            np.testing.assert_array_equal(
+                predictor_warm.nodes,
+                predictor_no_warm.nodes
+            )
+
+    # Check identical predictions
+    np.testing.assert_allclose(gb_warm_start.predict(X),
+                               gb_no_warm_start.predict(X))
+
+
+@pytest.mark.parametrize('GradientBoosting, X, y', [
+    (HistGradientBoostingClassifier, X_classification, y_classification),
+    (HistGradientBoostingRegressor, X_regression, y_regression)
+])
+def test_warm_start_max_depth(GradientBoosting, X, y):
+    """I did not understand what you expect."""
+    pass
+
+
+@pytest.mark.parametrize('GradientBoosting, X, y', [
+    (HistGradientBoostingClassifier, X_classification, y_classification),
+    (HistGradientBoostingRegressor, X_regression, y_regression)
+])
+def test_warm_start_early_stopping(GradientBoosting, X, y):
+    """Test that early stopping works as expected with warm starting.
+
+    Make sure that early stopping occurs after a small number of iterations
+    when fitting a second time with warm starting.
+
```

Comment:
You don't need blank spaces at the end of docstrings. Also nit: for tests we usually just use comments instead of `"""..."""` (don't know why)

Reply:
Okay, it's just that my linter was not very happy about it. I'm going to replace the docstrings with comments :)